src/components/features/main/ZonesContentsTable.tsx

```diff
@@ -41,31 +51,39 @@ export default function ZonesContentsTable() {
       entryNumber: 1,
       vendorName: 'truck',
       receptionNumbers: 12,
-      numberOfGoods: 33,
-      status: 'finished',
-      actions: (
-        <TableActionsMenu
-          itemProps={['MoveToNewZone', 'StartProcessing', 'FinishProcessing', 'DeliveryDetails']}
-          page="zones"
-        />
-      )
+      quantity: 33,
+      status: 'finished'
     },
     {
       entryNumber: 2,
       vendorName: 'truck',
       receptionNumbers: 11,
-      numberOfGoods: 52,
-      status: 'processing',
+      quantity: 52,
+      status: 'processing'
+    }
+  ]
+
+  const tableRowDataWithActions = rowData.map((row) => {
+    return {
+      ...row,
       actions: (
         <TableActionsMenu
-          itemProps={['MoveToNewZone', 'StartProcessing', 'FinishProcessing', 'DeliveryDetails']}
-          page="zones"
+          specificOptionHandler={(action: string) =>
+            onOpenMoveEntryDialog(action, row.quantity)
+          }
+          itemProps={[
+            ZonesTableActions.MoveToNewZone,
+            ZonesTableActions.StartProcessing,
+            ZonesTableActions.FinishProcessing,
+            ZonesTableActions.DeliveryDetails
```

Comment:
we can maybe have am array of objects which looks like this:
```js
const zonesTableActions = [
{
    title: 'zones.translations.translation...' // title is the translations string'
    value: ZonesTableActions.MoveToNewZone
}
// other options
]
```

this way we can remove the `page` prop since i feel like it complicates the `TableActionsMenu` component and in `TableActionsMenu` we use the `title` for the text and the value when we select an item.

Reply:
I am using page to dynamicly set the `title` and the `content` for <WarningActionDialog/> in <TableActionsMenu/>
![image](https://github.com/InterCarsBulgaria-TechLeadAcademy/warehouse.management.fe/assets/50380455/be0b959e-c887-480e-9e62-f12b44fcb29b)
If you want to remove the page from props we can find another solution, for example to include page name in every object in the array in itemProps and jut to take first object in the array and set the name of the page where I need. Or something similar, we can discuss it in discord if you want?